internal/graph/taskrunner.go

```diff
@@ -50,13 +49,9 @@ func NewTaskRunner(ctx context.Context, concurrencyLimit uint16) *TaskRunner {
 // Schedule schedules a task to be run. This is safe to call from within another
 // task handler function and immediately returns.
 func (tr *TaskRunner) Schedule(f TaskFunc) {
-	tr.wg.Add(1)
-
-	tr.lock.Lock()
-	tr.tasks = append(tr.tasks, f)
-	tr.lock.Unlock()
-
-	tr.spawnIfAvailable()
+	if tr.addTask(f) {
```

Comment:
why silently ignoring? `addTask` could return `error` instead and `Schedule` as well, hitting that we should not longer continue scheduling work

Reply:
The interface for `Schedule` is a "fire and forget". The error itself will be returned by the `Wait` command